### ISSUE TITLE
Getting a Segmentation Fault When Running update_abstract_interp_lib.sh

### ISSUE THREAD

**Issue by zjgarvey** [llvm/torch-mlir#2746]
Following the documentation in [development.md](https://github.com/llvm/torch-mlir/blob/main/docs/development.md), I have a fresh build of torch-mlir with the following cmake options: 

```bash
cmake -GNinja -Bbuild \
  -DCMAKE_BUILD_TYPE=RelWithDebInfo \
  -DPython3_FIND_VIRTUALENV=ONLY \
  -DLLVM_ENABLE_PROJECTS=mlir \
  -DLLVM_EXTERNAL_PROJECTS="torch-mlir" \
  -DLLVM_EXTERNAL_TORCH_MLIR_SOURCE_DIR="$PWD" \
  -DMLIR_ENABLE_BINDINGS_PYTHON=ON \
  -DLLVM_ENABLE_ASSERTIONS=ON \
  -DLLVM_TARGETS_TO_BUILD=host \
  -DCMAKE_C_COMPILER=clang -DCMAKE_CXX_COMPILER=clang++ \
  externals/llvm-project/llvm
  ```
  
  Running

```bash
build_tools/update_abstract_interp_lib.sh
```
gives a segmentation fault.

```
/home/zjgar/code/torch-mlir/build_tools/update_abstract_interp_lib.sh: line 47: 328861 Segmentation fault      PYTHONPATH="${pypath}" python -m torch_mlir.jit_ir_importer.build_tools.abstract_interp_lib_gen --pytorch_op_extensions=${ext_module:-""} --torch_transforms_cpp_dir="${torch_transforms_cpp_dir}"
```

This is the python file I ran through gdb:

```
gdb --args python /home/zjgar/code/torch-mlir/build/tools/torch-mlir/python_packages/torch_mlir/torch_mlir/jit_ir_importer/build_tools/abstract_interp_lib_gen.py
```

Here is the backtrace:

```
#0  0x00000000005266a0 in ?? ()
#1  0x0000000000504e03 in _PyEval_EvalFrameDefault ()
#2  0x000000000062e1b4 in ?? ()
#3  0x00000000004f3a67 in PyEval_EvalCode ()
#4  0x0000000000569033 in ?? ()
#5  0x000000000050d977 in ?? ()
#6  0x00000000004fefe1 in _PyEval_EvalFrameDefault ()
#7  0x0000000000531823 in _PyFunction_Vectorcall ()
#8  0x000000000051a2ae in ?? ()
#9  0x000000000055bfb1 in PyObject_CallMethodObjArgs ()
#10 0x000000000055a8d8 in PyImport_ImportModuleLevelObject ()
#11 0x0000000000500173 in _PyEval_EvalFrameDefault ()
#12 0x000000000062e1b4 in ?? ()
#13 0x00000000004f3a67 in PyEval_EvalCode ()
#14 0x0000000000647c37 in ?? ()
#15 0x0000000000645350 in ?? ()
#16 0x0000000000650d15 in ?? ()
#17 0x0000000000650a64 in _PyRun_SimpleFileObject ()
#18 0x0000000000650833 in _PyRun_AnyFileObject ()
#19 0x000000000064f787 in Py_RunMain ()
#20 0x000000000061ee0d in Py_BytesMain ()
#21 0x00007ffff7c82d90 in __libc_start_call_main (main=main@entry=0x61ed60, argc=argc@entry=2, argv=argv@entry=0x7fffffffd768) at ../sysdeps/nptl/libc_start_call_main.h:58
#22 0x00007ffff7c82e40 in __libc_start_main_impl (main=0x61ed60, argc=2, argv=0x7fffffffd768, init=<optimized out>, fini=<optimized out>, rtld_fini=<optimized out>, stack_end=0x7fffffffd758)
```

My python version is Python 3.11.0rc1 running on WSL: Ubuntu.

**Comment by zjgarvey** [comment]
I've found out what caused the segmentation fault in that script. Fascinatingly enough, the following line by itself causes a segmentation fault for me!

```python
import torch.jit._shape_functions
```

I'll do some looking around. If anyone knows why this might be the case please let me know.

**Comment by zjgarvey** [comment]
I've messed around with my python build a little bit.

If anyone wants to reproduce the segmentation fault on a fresh ubuntu-22.04:

```bash
sudo apt update
sudo apt install python3.11
sudo apt install python3.11-venv
python3.11 -m venv test_venv
source test_venv/bin/activate
pip install --upgrade pip
pip install numpy
pip install torch
python3.11
```

then 

```python
import torch.jit._shape_functions
```

will result in a segmentation fault. 

Importing that module on Ubuntu's default python3 version 3.10.6 does not reproduce the segmentation fault. 

I found that building python version 3.11.7 from source and using the venv package included there avoids the segmentation fault as well. This is probably the preferred solution, if anyone else encounters this issue in the future.
